publish.yml

```diff
@@ -0,0 +1,34 @@
+name: Publish on Docker Hub
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          username: ${{ secrets.DOCKER_LOGIN_USERNAME }}
+          password: ${{ secrets.DOCKER_LOGIN_PASSWORD }}
+      
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: my-docker-hub-namespace/my-docker-hub-repository
```

Comment:
@rscohn2  whoops; this should be intelpython/intelpython3_core. Should it be:

    images:
        - intelpython/intelpython3_core
        - intelpython/intelpython3_full

Reply:
The old script pushed multiple tags for the same image. If the github tag is 2017.0.0-1, then the docker tag would be 2017.0.0-1, 2017.0, and latest. Looking at the documentation for the metadata-action, I think it will push 2017.0.0-1 and latest. The semver option https://github.com/docker/metadata-action#semver can extract the major.minor tag automatically, but our tags are not semver because of the -1 on the end. Choices are:
* stop tagging with major.minor. this is easiest and probably ok.
* manually put in the tags for every commit
* generate the tags by using the template processor to generate publish.ytml
* when the conda label is 2017.0.0-1, tag the repo with 2017.0.1

Reply:
@rscohn2 I think option 1 is best as well. There shouldn't be a 2017.0 tag if there's a chance it would have to be updated because of a CVE fix. Users can just rely on 'latest' and the snapshot itself (major.min.patch-build)